tests/models/test_baseoperator.py

```diff
@@ -262,6 +263,33 @@ def test_email_on_actions(self):
         assert test_task.email_on_retry is False
         assert test_task.email_on_failure is True
 
+    def test_upstream_is_set_when_template_field_is_xcomarg(self):
+        class CustomOpSuperBefore(DummyOperator):
+            template_fields = ("field",)
+
+            @apply_defaults
+            def __init__(self, field, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+                self.field = field
+
+        class CustomOpSuperAfter(DummyOperator):
+            template_fields = ("field",)
+
+            @apply_defaults
+            def __init__(self, field, *args, **kwargs):
+                self.field = field
+                super().__init__(*args, **kwargs)
+
+        with DAG("test_dag", default_args={"start_date": datetime.today()}):
+            op1 = DummyOperator(task_id="op1")
+            op2 = CustomOpSuperBefore(task_id="op2", field=op1.output)
+            op3 = CustomOpSuperAfter(task_id="op3", field=op1.output)
+
+        assert op1 in op2.upstream_list
+        assert op1 in op3.upstream_list
+        assert op2 in op1.downstream_list
+        assert op3 in op1.downstream_list
+
```

Comment:
we need checks on "operator defined outside of DAG",  pythonOperator (op_kwargs) usage

Reply:
Added